tools/smufl/smufl_fontgen.js

```diff
@@ -11,40 +8,54 @@ const opentype = require('opentype.js');
 
 function LogError(...args) {
   // eslint-disable-next-line
-  console.error(...args)
+  console.error(...args);
 }
 
-// Converte OTF glyph path to Vexflow glyph path
+// Convert OTF glyph path to Vexflow glyph path.
 function toVFPath(glyph) {
+  // TODO: What if we convert it to 1000 pt???
```

Comment:
I think the 72 here comes from the fact that there are 72 points to an inch.

Ideally, the size passed here should be as close as possible to the actual size that the glyphs will be rendered. This is because the font rendering engine will fit the curve data to the pixel grid to prevent strokes being too thin to display. There's more about it here: https://docs.microsoft.com/en-us/typography/opentype/spec/ttch01#grid-fitting-a-glyph-outline

Reply:
I think this is the most important part of the discussion.

What font size should we extract the glyphs at? The reason I chose 1000 is because the OTFs are designed on a 1000 x 1000 grid. As a result, all the paths come out as integers and look super clean and nice.

However, the final paths have transforms that look like `translate(17 85) scale(0.02808 0.02808)`. So we are scaling noteheads to 2.8% of its 1000pt size (so maybe around 28 pt).

Is there a good way to measure A) the performance impact and B) the rendering quality of scaling paths down to 2.8%?

Or should we pre-scale and pre-round the paths (to 3 decimal points) so that our final transforms look more like `translate(17 85) scale(0.5 0.5)`?

But what if a developer wants to draw at 4x or 8x or 16x size (for folks with poor visibility)?

Maybe we need to run a test. Ideally there is no performance difference, because we are drawing the same number of points, and just multiplying the path by a transform matrix that includes a translate and a scale.

Any ideas on this front? I do personally really really like the look of the all-integer paths (extracted at 1000 pt). :-)

Reply:
I think I'm beginning to understand what you're saying. So we are using the opentype.js library to extract these paths. If we extract the paths at say, 36 pt, would the opentype.js library use font hinting or some other process to make the paths look best at 36 pt?

Hopefully the Bravura & Petaluma & Gonville designers didn't make any super thin lines at 1000 x 1000 that would disappear when I do a naive scale down to 2-5%.

I think we'd need to extract sets of paths at different pt sizes to see if there are any troublesome glyphs.

Reply:
Yep, I think the only way to really know for sure is to try it and see what they look like extracted at 1000 x 1000.